Flyway documentation should mention required database modules

Trying to use Flyway, I imported `quarkus-flyway` and then it complained about:

```
Caused by: org.flywaydb.core.api.FlywayException: Unsupported Database: PostgreSQL 16.3
	at org.flywaydb.core.internal.database.DatabaseTypeRegister.getDatabaseTypeForConnection(DatabaseTypeRegister.java:105)
	at org.flywaydb.core.internal.jdbc.JdbcConnectionFactory.<init>(JdbcConnectionFactory.java:73)
	at org.flywaydb.core.FlywayExecutor.execute(FlywayExecutor.java:134)
	at org.flywaydb.core.Flyway.migrate(Flyway.java:147)
	at io.quarkus.flyway.runtime.FlywayRecorder.doStartActions(FlywayRecorder.java:136)
	at io.quarkus.deployment.steps.FlywayProcessor$startActions2099152139.deploy_0(Unknown Source)
	at io.quarkus.deployment.steps.FlywayProcessor$startActions2099152139.deploy(Unknown Source)
```

The documentation at https://quarkus.io/guides/flyway is not clear, because it does mention that other DBs need special extra modules, but it doesn't say that about Postgres. Perhaps it's new, but now it's required, so we should document that:

```
<dependency>
    <groupId>org.flywaydb</groupId>
    <artifactId>flyway-database-postgresql</artifactId>
</dependency>
```

Is now required. Also, I'm a bit suspicious about the fact that some modules start with `flyway-database` and others just `flyway-` I don't know if that's really the case, we should check.

And finally, but VERY IMPORTANT, we should fix that stupid exception listed above, because it's NOT helpful. I installed other versions of Postgres, I looked online to find which version was supported, and got really mixed results from Flyway documentation, one page saying only 16, another saying all of them.

It was not that my DB version was unsupported, the real error was that the extra module was missing. If we can't automatically add those modules at build time (if flyway + postgres -> extra dependency) (ask @aloubyansky) then perhaps we should throw a build time error with helpful text for how to resolve it by adding the required module?

I just want to reiterate @FroMage's initial point that the documentation should be updated, I was following the [official Quarkus' Using Flyway guide](https://quarkus.io/guides/flyway#setting-up-support-for-flyway) and by following exactly what is documented for Postgres you would get the above error.
In my case, since I'm using the dev-services provided postgres instance, I got the following:
```kotlin
 Failed to start quarkus: io.quarkus.dev.appstate.ApplicationStartException: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.dev.appstate.ApplicationStateNotification.waitForApplicationStart(ApplicationStateNotification.java:58)
	at io.quarkus.runner.bootstrap.StartupActionImpl.runMainClass(StartupActionImpl.java:132)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.restartApp(IsolatedDevModeMain.java:193)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.restartCallback(IsolatedDevModeMain.java:174)
	at io.quarkus.deployment.dev.RuntimeUpdatesProcessor.doScan(RuntimeUpdatesProcessor.java:548)
	at io.quarkus.deployment.dev.RuntimeUpdatesProcessor.doScan(RuntimeUpdatesProcessor.java:448)
	at io.quarkus.vertx.http.runtime.devmode.VertxHttpHotReplacementSetup$6.call(VertxHttpHotReplacementSetup.java:161)
	at io.quarkus.vertx.http.runtime.devmode.VertxHttpHotReplacementSetup$6.call(VertxHttpHotReplacementSetup.java:148)
	at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$0(ContextImpl.java:178)
	at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:279)
	at io.vertx.core.impl.ContextImpl.lambda$internalExecuteBlocking$2(ContextImpl.java:210)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2516)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2495)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1495)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:11)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:11)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
	at io.quarkus.runtime.Application.start(Application.java:101)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:111)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:113)
	... 1 more
Caused by: org.flywaydb.core.api.FlywayException: Unsupported Database: PostgreSQL 14.12
	at org.flywaydb.core.internal.database.DatabaseTypeRegister.lambda$getDatabaseTypeForConnection$7(DatabaseTypeRegister.java:122)
	at java.base/java.util.Optional.orElseThrow(Optional.java:403)
	at org.flywaydb.core.internal.database.DatabaseTypeRegister.getDatabaseTypeForConnection(DatabaseTypeRegister.java:122)
	at org.flywaydb.core.internal.jdbc.JdbcConnectionFactory.<init>(JdbcConnectionFactory.java:77)
	at org.flywaydb.core.FlywayExecutor.execute(FlywayExecutor.java:138)
	at org.flywaydb.core.Flyway.clean(Flyway.java:273)
	at io.quarkus.flyway.runtime.FlywayRecorder.doStartActions(FlywayRecorder.java:123)
	at io.quarkus.deployment.steps.FlywayProcessor$startActions2099152139.deploy_0(Unknown Source)
	at io.quarkus.deployment.steps.FlywayProcessor$startActions2099152139.deploy(Unknown Source)
	... 11 more
```
Only after digging around a bit and finding [this issue on flyway's Github](https://github.com/flyway/flyway/issues/3780) did I manage to get it working.

--- a/docs/src/main/asciidoc/flyway.adoc
+++ b/docs/src/main/asciidoc/flyway.adoc
@@ -28,9 +28,7 @@ In your build file, add the following dependencies:
 
 * the Flyway extension
 * your JDBC driver extension (`quarkus-jdbc-postgresql`, `quarkus-jdbc-h2`, `quarkus-jdbc-mariadb`, ...)
-* the MariaDB/MySQL support is now in a separate dependency, MariaDB/MySQL users need to add the `flyway-mysql` dependency from now on.
-* the Microsoft SQL Server support is now in a separate dependency, Microsoft SQL Server users need to add the `flyway-sqlserver` dependency from now on.
-* the Oracle support is now in a separate dependency, Oracle users need to add the `flyway-database-oracle` dependency from now on.
+* unless you're using in-memory or file databases (such as H2 or SQLite), you need to add a flyway module dependency corresponding to the database you're using. (https://github.com/flyway/flyway/issues/3780[for more details])
 
 [source,xml,role="primary asciidoc-tabs-target-sync-cli asciidoc-tabs-target-sync-maven"]
 .pom.xml
@@ -39,6 +37,12 @@ In your build file, add the following dependencies:
 <dependency>
     <groupId>io.quarkus</groupId>
     <artifactId>quarkus-flyway</artifactId>
+</dependency>
+
+<!-- JDBC driver dependencies -->
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-jdbc-postgresql</artifactId>
 </dependency>
 
 <!-- Flyway SQL Server specific dependencies -->
@@ -59,11 +63,84 @@ In your build file, add the following dependencies:
     <artifactId>flyway-database-oracle</artifactId>
 </dependency>
 
-<!-- JDBC driver dependencies -->
+<!-- Flyway Postgres specific dependencies -->
 <dependency>
-    <groupId>io.quarkus</groupId>
-    <artifactId>quarkus-jdbc-postgresql</artifactId>
+    <groupId>org.flywaydb</groupId>
+    <artifactId>flyway-database-postgresql</artifactId>
 </dependency>
+
+<!-- Flyway DB2 specific dependencies -->
+<dependency>
+    <groupId>org.flywaydb</groupId>
+    <artifactId>flyway-database-db2</artifactId>
+</dependency>
+
+<!-- Derby specific dependencies -->
+<dependency>
+    <groupId>org.flywaydb</groupId>
+    <artifactId>flyway-database-derby</artifactId>
+</dependency>
+
+<!-- HSQLDB specific dependencies -->
+<dependency>
+    <groupId>org.flywaydb</groupId>
+    <artifactId>flyway-database-hsqldb</artifactId>
+</dependency>
+
+<!-- Informix specific dependencies -->
+<dependency>
+    <groupId>org.flywaydb</groupId>
+    <artifactId>flyway-database-informix</artifactId>
+</dependency>
+
+<!-- Redshift specific dependencies -->
+<dependency>
+    <groupId>org.flywaydb</groupId>
+    <artifactId>flyway-database-redshift</artifactId>
+</dependency>
+
+<!-- Saphana specific dependencies -->
+<dependency>
+    <groupId>org.flywaydb</groupId>
+    <artifactId>flyway-database-saphana</artifactId>
+</dependency>
+
+<!-- Snowflake specific dependencies -->
+<dependency>
+    <groupId>org.flywaydb</groupId>
+    <artifactId>flyway-database-snowflake</artifactId>
+</dependency>
+
+<!-- Sybasease specific dependencies -->
+<dependency>
+    <groupId>org.flywaydb</groupId>
+    <artifactId>flyway-database-sybasease</artifactId>
+</dependency>
+
+<!-- Firebird specific dependencies -->
+<dependency>
+    <groupId>org.flywaydb</groupId>
+    <artifactId>flyway-firebird</artifactId>
+</dependency>
+
+<!-- BigQuery specific dependencies -->
+<dependency>
+    <groupId>org.flywaydb</groupId>
+    <artifactId>flyway-gcp-bigquery</artifactId>
+</dependency>
+
+<!-- Spanner specific dependencies -->
+<dependency>
+    <groupId>org.flywaydb</groupId>
+    <artifactId>flyway-gcp-spanner</artifactId>
+</dependency>
+
+<!-- Singlestore specific dependencies -->
+<dependency>
+    <groupId>org.flywaydb</groupId>
+    <artifactId>flyway-singlestore</artifactId>
+</dependency>
+
 ----
 
 [source,gradle,role="secondary asciidoc-tabs-target-sync-gradle"]
@@ -71,14 +148,41 @@ In your build file, add the following dependencies:
 ----
 // Flyway specific dependencies
 implementation("io.quarkus:quarkus-flyway")
+// JDBC driver dependencies
+implementation("io.quarkus:quarkus-jdbc-postgresql")
 // Flyway SQL Server specific dependencies
 implementation("org.flywaydb:flyway-sqlserver")
 // Flyway MariaDB/MySQL specific dependencies
 implementation("org.flywaydb:flyway-mysql")
 // Flyway Oracle specific dependencies
 implementation("org.flywaydb:flyway-database-oracle")
-// JDBC driver dependencies
-implementation("io.quarkus:quarkus-jdbc-postgresql")
+// Flyway Postgres specific dependencies
+implementation("org.flywaydb:flyway-database-postgresql")
+// Flyway DB2 specific dependencies
+implementation("org.flywaydb:flyway-database-db2")
+// Flyway Derby specific dependencies
+implementation("org.flywaydb:flyway-database-derby")
+// HSQLDB specific dependencies
+implementation("org.flywaydb:flyway-database-hsqldb")
+// Informix specific dependencies
+implementation("org.flywaydb:flyway-database-informix")
+// Redshift specific dependencies 
+implementation("org.flywaydb:flyway-database-redshift")
+// Saphana specific dependencies
+implementation("org.flywaydb:flyway-database-saphana")
+// Snowflake specific dependencies
+implementation("org.flywaydb:flyway-database-snowflake")
+// Sybasease specific dependencies
+implementation("org.flywaydb:flyway-database-sybasease")
+// Firebird specific dependencies
+implementation("org.flywaydb:flyway-firebird")
+// BigQuery specific dependencies
+implementation("org.flywaydb:flyway-gcp-bigquery")
+// Spanner specific dependencies
+implementation("org.flywaydb:flyway-gcp-spanner")
+// Singlestore specific dependencies
+implementation("org.flywaydb:flyway-singlestore:10.15.0")
+
 ----
 
 Flyway support relies on the Quarkus datasource config.

